src/geometry/manifold/charted_space.lean

```diff
@@ -666,6 +747,38 @@ end
 
 end maximal_atlas
 
+section singleton
+variables {α : Type*} [topological_space α]
+variables {e : local_homeomorph α H}
+
+/-- If a single local homeomorphism `e` from a space `α` into `H` has source covering the whole
+space `α`, then that local homeomorphism induces an `H`-charted space structure on `α`.
+(This condition is equivalent to `e` being an open embedding of `α` into `H`.) -/
+def singleton_charted_space (h : e.source = set.univ) : charted_space H α :=
```

Comment:
As far as I understand, Lean will print this as `singleton_charted_space _`. If you make `e` an explicit argument, then it will look like `singleton_charted_space e _`.

Reply:
Which do you recommend?  `e` can be inferred from the hypothesis `h`, right?

Reply:
It's just for pretty-printing purposes: in the goal state, it is nice to see what you are speaking about, and Prop arguments are hidden (replaced with `_`) since they are irrelevant.